src/Microsoft.AspNetCore.Http.Extensions/UriHelper.cs

```diff
@@ -42,7 +42,7 @@ public static class UriHelper
         /// <param name="query"></param>
         /// <param name="fragment"></param>
         /// <returns></returns>
-        public static string Encode(
+        public static string BuildAbsolute(
             string scheme,
             HostString host,
```

Comment:
Might make sense to add an overload that takes a string variation of this. Almost every call into this ends up doing `new HostString(...)`